### ISSUE TITLE
VS 2022 Windows Forms Designer gives 'Access is denied' error when using .Net 6.0 template

### ISSUE THREAD

**Issue by oyundev** [dotnet/winforms#6771]
* .NET Core Version:  **.NET SDKs installed: 6.0.200**

* Have you experienced this same bug with .NET Framework?:  **No**

**Problem description:**
```
VS2022 Windows Forms designer giving 'Access is denied' error when 
I try to open form designer in a fresh project created by .Net 6.0 LTS Forms template. 
Interestingly, if I run VS with administrative rights, this error does not occur. 
This problem does not occur when using projects created by 
.Net Framework 4.8 template or WPF template.
```

**Expected behavior:**

```
I assume this is not the expected behaviour.
```
**Minimal repro:**

![vserr](https://user-images.githubusercontent.com/44315345/155897994-c757bd4c-ea79-4ed4-a683-e3923c8cdc4b.png)
![vstmplt](https://user-images.githubusercontent.com/44315345/155897995-e58c98a5-f890-4d0b-a0fd-902ff2ef2b91.png)

## Stack trace
```
Access is denied    
Instances of this error (1)  
 
1.   Hide Call Stack 
 
at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
at Microsoft.DotNet.DesignTools.Client.Host.ServerProcess.LaunchServerProcess(UInt32 lcid, Boolean hideMainWindow, Boolean enableCoreHostTrace, Boolean autoAttachDebugger, String pipeName)
at Microsoft.DotNet.DesignTools.Client.Host.ServerProcess.d__25.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Microsoft.DotNet.DesignTools.Client.Host.ServerHostFactory.d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Microsoft.DotNet.DesignTools.Client.DesignToolsClientLoader.d__28.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at Microsoft.DotNet.DesignTools.Client.DesignToolsClientLoader.d__28.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Microsoft.DotNet.DesignTools.Client.DesignToolsClientLoader.<>c__DisplayClass24_1.<<-ctor>b__1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Microsoft.DotNet.DesignTools.Client.DesignToolsClientLoader.d__26.MoveNext()  
```
 

## Version 
```
Microsoft Visual Studio Community 2022 (64-bit) 
Version 17.1.0, Windows 11 Pro (22000.527)
```

## .csproj
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net6.0-windows</TargetFramework>
    <Nullable>enable</Nullable>
    <UseWindowsForms>true</UseWindowsForms>
    <ImplicitUsings>enable</ImplicitUsings>
  </PropertyGroup>

</Project>
```

##dotnet --info
```
C:\>dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.200
 Commit:    4c30de7899

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.200\

Host (useful for support):
  Version: 6.0.2
  Commit:  839cdfb0ec

.NET SDKs installed:
  6.0.200 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by RussKie** [comment]
Please enable logging and see if you can find more info.

**Comment by RussKie** [comment]
Please enable logging and see if you can find more info. https://github.com/dotnet/winforms/blob/main/docs/designer/troubleshooting.md

**Comment by oyundev** [comment]
Today, I tested to create a project with windows forms .Net 6 template on another machine, Windows 10 + recenty installed VS 2022 and it just worked without any problems at all. 

Thinking this might be a issue related to target directory permissions so this time I chose another location under %userprofile%\source\repos but I got the same error.

Changed to Verbose logging at Options->Windows Forms Designer->.Net Core->Logging Level

Debug output:
```
[16:39:59.7709449] Added 'System.Windows.Forms'
[16:39:59.7719482] Loaded System in 00:00:00.0000075 [thread 57]
[16:39:59.7719482] Processed 'System' in 00:00:00.0006268
[16:39:59.7855088] Added 'System'
[16:39:59.7855088] Loaded System.Drawing in 00:00:00.0000112 [thread 57]
[16:39:59.7855088] Processed 'System.Drawing' in 00:00:00.0001014
[16:39:59.7875225] Added 'System.Drawing'
[16:39:59.7875225] Loaded System.Design in 00:00:00.0000080 [thread 57]
[16:39:59.7925095] Processed 'System.Design' in 00:00:00.0051632
[16:39:59.7925095] Creating VSCodeDomDesignerLoader
[16:39:59.8010275] Added 'System.Design'
[16:39:59.8020305] Loaded System.Drawing.Design in 00:00:00.0000174 [thread 57]
[16:39:59.8020305] Processed 'System.Drawing.Design' in 00:00:00.0002151
[16:39:59.8020305] Added 'System.Drawing.Design'
[16:39:59.8562231] [Thread 1] Creating design-time ITypeResolutionService.
[16:39:59.8713742] Start processing enqueued assemblies
[16:39:59.8713742] Loaded Microsoft.WinForms.DesignTools.Protocol in 00:00:00.0000150 [thread 52]
[16:39:59.8773758] Processed 'Microsoft.WinForms.DesignTools.Protocol' in 00:00:00.0064165
[16:39:59.8813770] Added 'Microsoft.WinForms.DesignTools.Protocol'
[16:39:59.8813770] Loaded Microsoft.WinForms.DesignTools.Client in 00:00:00.0000122 [thread 52]
[16:39:59.8903763] Processed 'Microsoft.WinForms.DesignTools.Client' in 00:00:00.0085945
[16:39:59.8923764] Added 'Microsoft.WinForms.DesignTools.Client'
[16:39:59.9981161] [Thread 50] Launching server...
[16:40:00.8693254] [WinFormsApp1]: Preparing shadow cache folder
[16:40:00.8693254] [WinFormsApp1]: Copying design tools server common files from: 'c:\program files\microsoft visual studio\2022\community\common7\ide\commonextensions\microsoft\windows.forms\DesignToolsServer\Common'
[16:40:01.0358778] [WinFormsApp1]: Copying design tool server platform-specific files from: 'c:\program files\microsoft visual studio\2022\community\common7\ide\commonextensions\microsoft\windows.forms\DesignToolsServer\x64'
[16:40:01.0388778] [WinFormsApp1]: Copying 'C:\Users\ORHANT\source\repos\WinFormsApp1\obj\Debug\net6.0-windows\WinFormsApp1.designer.deps.json' to 'DesignToolsServer.deps.json'
[16:40:01.0388778] [WinFormsApp1]: Copying 'C:\Users\ORHANT\source\repos\WinFormsApp1\obj\Debug\net6.0-windows\WinFormsApp1.designer.runtimeconfig.json' to 'DesignToolsServer.runtimeconfig.json'
[16:40:01.1601901] [WinFormsApp1]: Launching design tools server process...
[16:40:01.1681822] System.ComponentModel.Win32Exception (0x80004005): Access is denied
                      at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
                      at Microsoft.DotNet.DesignTools.Client.Host.ServerProcess.LaunchServerProcess(UInt32 lcid, Boolean hideMainWindow, Boolean enableCoreHostTrace, Boolean autoAttachDebugger, String pipeName)
                      at Microsoft.DotNet.DesignTools.Client.Host.ServerProcess.<LaunchAsync>d__25.MoveNext()
                   --- End of stack trace from previous location where exception was thrown ---
                      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                      at Microsoft.DotNet.DesignTools.Client.Host.ServerHostFactory.<CreateHostAsync>d__7.MoveNext()
                   --- End of stack trace from previous location where exception was thrown ---
                      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                      at Microsoft.DotNet.DesignTools.Client.DesignToolsClientLoader.<CreateClientAsync>d__28.MoveNext()
                   --- End of stack trace from previous location where exception was thrown ---
                      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                      at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
[16:40:01.1933170] Cannot update selection in server process as Session is not connected.
```

**Comment by Shyam-Gupta** [comment]
@oyundev WinForms .NET (Core) Designer launches a child process, `DesignToolsServer.exe` for its operation. From the logs it appears that error occurs during process launch. When this issue occurs, could you please check if any error gets logged in `Windows EventVwr`. Steps:

- Press `Windows + R` to launch `Run` dialog
- Type `EventVwr` and press enter
- Select `Custom Views` -> `Administrative Events` node and check if any error has been logged for `DesignToolsServer.exe` process or for `devenv.exe` process

Also, could you please confirm if `WPF .NET (Core) designer` works fine when VS is **not launched** with administrative privilege:
![image](https://user-images.githubusercontent.com/24871676/156141031-8c9f101c-e7a2-4383-8a99-fbe0e63ca511.png)

**Comment by oyundev** [comment]
Already mentioned in description when I open this issue that this problem doesn't occur with WPF .Net template. I'm familiar with eventvwr.msc from NT4.0 days. Ahh, good old times. Anyway, looking there was my first instinct but unfortunately there was nothing related to DesignToolsServer.exe or devenv.exe in there. 

I used sysinternals procmon and find out that file permissions on file at temp location is incorrect somehow. Only admins had exec rights.(?)

![procmon](https://user-images.githubusercontent.com/44315345/156210784-00b3d3f9-bcaf-4d32-ab53-b816da0ea0b8.png)
![permsn](https://user-images.githubusercontent.com/44315345/156210797-5ffb5d77-35f2-4d44-96e2-0938c0580f91.png)

I tested with another fresh windows account and no problem, it just worked wonderfully! Under that new account profile my user had full access rights over the file at temp folder created by vs. 

My account profile seems to be the culprit. Meaning I can work with a new windows account profle without re-installing everything from scratch. I can live with that. I'm closing the issue.
